### DISCUSSION TITLE
How to apply velocity smooth constraint and weight

### DISCUSSION THREAD

**Discussion by dbdxnuliba** [stephane-caron/pink#91] [Q&A] [answered]
how could we aplly velocity smooth constraint and weight

**Comment by stephane-caron** [comment, marked as answer]
I see different ways to interpret your question, so here are some options depending on your use case:

- If by smooth velocity constraint you mean that velocities cannot be too high, you can use the [`VelocityLimit`](https://stephane-caron.github.io/pink/limits.html#pink.limits.VelocityLimit).
- If you rather mean that velocity *variations* should not be too high, then you can derive from this class to additionally store/pass a current velocity parameter, and constraint the maximum deviation of the new velocity from the current one.
- If by "weight" you refer to a penalization in the cost function (soft) rather than inequality constraints (hard), then you can take a look at [tasks](https://stephane-caron.github.io/pink/tasks.html) rather than [limits](https://stephane-caron.github.io/pink/limits.html).
- And if your use case is more complex, you can also take a look at [barriers](https://stephane-caron.github.io/pink/barriers.html), which are more general than limits and tasks.

Hoping this helps!
